src/test/signal/test_signals.rs

```diff
@@ -501,6 +501,37 @@ fn test_handled_kill_interrupts_syscall() -> Result<(), Box<dyn Error>> {
     Ok(())
 }
 
+fn test_restart() -> Result<(), Box<dyn Error>> {
+    let signal = Signal::SIGUSR1;
+    unsafe {
+        signal::sigaction(
+            signal,
+            &signal::SigAction::new(
+                signal::SigHandler::Handler(nop_signal_handler),
+                signal::SaFlags::SA_RESTART,
```

Comment:
What happens if there are two signal handlers, and only one `SA_RESTART`? It looks like shadow will restart if at least one signal handler is `SA_RESTART`, so maybe there should be a test for this?